docs/guide.rst

```diff
@@ -31,108 +31,131 @@ The `WGPU <https://github.com/pygfx/wgpu-py>`_ library provides the low level AP
+by defining an empty `Scene`::
 
+    import pygfx as gfx
 
-Opening a window to render to
------------------------------
+    scene = gfx.Scene()
 
-Your visualization must end up on the screen somehow. For this, we use the
-canvas abstraction provided by wgpu-py.
+Right now this scene is very desolate. There is no light, no objects, and
+nothing that can look at those objects. Let's change this by adding some
+`light` and a `camera`::
 
-.. code-block:: python
-
-    # Create Qt widget that can function as a canvas
-    from wgpu.gui.qt import WgpuCanvas
-    canvas = WgpuCanvas()
+    # and god said ...
+    scene.add(gfx.AmbientLight())
 
+    camera = gfx.PerspectiveCamera(70, 16 / 9)
+    camera.position.z = 400
 
-We can ask the canvas to schedule a draw event, and tell it what to do
-to perform a draw.
+Now there is light and a camera to perceive the light. To complete the setup
+we also need to add an object to look at::
 
-.. code-block:: python
+    geometry = gfx.box_geometry(200, 200, 200)
+    material = gfx.MeshPhongMaterial(color=(1, 1, 0, 1))
+    cube = gfx.Mesh(geometry, material)
 
-    def animate():
-       ...  # we'll get to this
+    scene.add(cube)
 
+Objects are slightly more complicated than lights or cameras. They have a
+`geometry`, which controlls an object's form, and a `material`, which controls
+an object's appearance (color, reflectiveness, etc).
 
-    canvas.request_draw(animate)
+Now we have all the necessary ingredients and it is time to take a look. To do
+so we need to create a `canvas` to draw what we see (here a on-screen window)
+and a `renderer` that will look at the scene (through the camera) and draw what
+it sees onto the canvas::
 
+    from wgpu.gui.auto import WgpuCanvas, run
 
-Setting up a renderer
----------------------
+    renderer = gfx.renderers.WgpuRenderer(WgpuCanvas())
+    renderer.render(scene, camera)
 
-To render your scene to the canvas, you need a renderer. A renderer
-needs a target to render to (which can be a canvas or a texture).
-And finally, you need a camera to define the point of view to render the scene from.
+    run()
 
-.. code-block:: python
+.. image:: _static/guide_static_cube.png
 
-    # A renderer is associated with a canvas (or a texture) that it renders to
-    renderer = gfx.renderers.WgpuRenderer(canvas)
+Nice, we rendered our first scene! While this technically completes the hello
+world example, we can go one step further and add a bit of flare. We can add a
+`draw_function` which allows us to add custom logic into the rendering process::
 
-    # A camera defines the viewpoint and projection
-    camera = gfx.PerspectiveCamera(70, 16 / 9)
+    def animate():
+        # rotate the cube a bit
+        rot = gfx.linalg.Quaternion().set_from_euler(gfx.linalg.Euler(0.005, 0.01))
+        cube.rotation.multiply(rot)
+        # update the canvas
+        renderer.render(scene, camera)
+        renderer.request_draw()
 
-    ...
+    # make the cube pop
```

Comment:
Pop may be confusion. Maybe something like:
```suggestion
    # Add a directional light to make the cube look better
```

Reply:
😅